File: src/com/rian/difficultycalculator/attributes/HighStrainSection.java
package com.rian.difficultycalculator.attributes;

/**
 * Represents a beatmap section at which the strains of objects are considerably high.
 */
public final class HighStrainSection {
    /**
     * The index of the first object in this section with respect to the full beatmap.
     */
    public final int firstObjectIndex;

    /**
     * The index of the last object in this section with respect to the full beatmap.
     */
    public final int lastObjectIndex;

    /**
     * The summed strain of this section.
     */
    public final double sumStrain;

    /**
     * @param firstObjectIndex The index of the first object in this section with respect to the full beatmap.
     * @param lastObjectIndex The index of the last object in this section with respect to the full beatmap.
     * @param sumStrain The summed strain of this section.
     */
    public HighStrainSection(int firstObjectIndex, int lastObjectIndex, double sumStrain) {
        this.firstObjectIndex = firstObjectIndex;
        this.lastObjectIndex = lastObjectIndex;
        this.sumStrain = sumStrain;
    }
}

File: src/com/rian/osu/difficulty/attributes/HighStrainSection.kt
package com.rian.osu.difficulty.attributes

import com.rian.osu.beatmap.Beatmap
import com.rian.osu.beatmap.hitobject.HitObject

/**
 * Represents a [Beatmap]'s section at which the strains of [HitObject]s are considerably high.
 */
data class HighStrainSection(
    /**
     * The index of the first [HitObject] in this [HighStrainSection] with respect to the full [Beatmap].
     */
    @JvmField
    val firstObjectIndex: Int,

    /**
     * The index of the last [HitObject] in this [HighStrainSection] with respect to the full [Beatmap].
     */
    @JvmField
    val lastObjectIndex: Int,

    /**
     * The summed strain of this [HighStrainSection].
     */
    @JvmField
    val sumStrain: Double
)